pkg/i2gw/providers/apisix/http_to_https.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apisix
+
+import (
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+func httpToHttps(ingressResources i2gw.InputResources, gatewayResources *i2gw.GatewayResources) field.ErrorList {
+	var errs field.ErrorList
+	for _, ingress := range ingressResources.Ingresses {
```

Comment:
I suggest you take inspiration from other already existing features on hot to retrieve routes and apply changes to them: https://github.com/kubernetes-sigs/ingress2gateway/blob/c55e274d9ee7f9ae2e02b21199495537060b0d8a/pkg/i2gw/providers/kong/method_matching.go#L38-L55. The way you implemented leave some aspects uncovered, as the ingress rules are not properly grouped.

Reply:
Updated.